Mage/src/main/java/mage/abilities/mana/ManaOptions.java

```diff
@@ -648,4 +560,32 @@ public String toString() {
             sb.append(',').append(' ');
         }
     }
+
+    /**
+     * Utility function to get a Mana from ManaOptions at the specified position.
+     * Since the implementation uses a LinkedHashSet the ordering of the items is preserved.
+     *
+     * NOTE: Do not use in tight loops as performance of the lookup is much worse than
+     *       for ArrayList (the previous superclass of ManaOptions).
+     *
+     * NOTE: Deprecated warning is there as a reminder for when this gets called that it's not
+     *       the usual get() function from an array.
+     *
+     */
+    @Deprecated
+    public Mana getAtIndex(int i) {
```

Comment:
Maybe name method something more unique like `getManaAtIndex` rather than adding the `Deprecated` tag to a method that isn't deprecated?